examples/vault-cluster-authentication-gce/README.md

```diff
@@ -1,36 +1,173 @@
+if an authentication source is a legitimate source or not.
 
-For more info on how the Vault cluster works, check out the [vault-cluster](https://github.com/hashicorp/terraform-google-vault/tree/master/modules/vault-cluster) documentation.
+There are currently two ways a GCP resource can authenticatate to Vault: `gce` and `iam`.
+In this example, we demonstrate the [GCP GCE Auth Method][gce_auth].
 
+For more info on how the Vault cluster works, check out the [vault-cluster][vault_cluster]
+documentation. For an example on using the `iam` method, check out the
+[vault-authentication-iam example][iam_example].
 
-## Quick start
+**Note**: This example launches a private vault cluster, meaning the nodes do not
+have public IP addresses and cannot talk to the outside world. If you need to SSH
+to your Vault cluster, check the [vault-cluster-private example][private_vault]
+for instructions on how to launch a Bastion host in the same subnet and use it to
+access the cluster.
+
+## Running this example
 
-To deploy a Vault Cluster:
+You will need to create a [Google Image][google_image] that has Vault and Consul
+installed, which you can do using the [vault-consul-image example][image_example].
+All the GCE Instances in this example (including the GCE Instance that authenticates
+to Vault) install [Dnsmasq][dnsmasq] (via the [install-dnsmasq module][dnsmasq_module])
+so that all DNS queries for `*.consul` will be directed to the Consul Server cluster.
+Because Consul has knowledge of all the Vault nodes (and in some cases, of other
+services as well), this setup allows the instances to use Consul's DNS server for
+service discovery, and thereby to discover the IP addresses of the Vault nodes.
+
+## Quick start
 
 1. `git clone` this repo to your computer.
-1. Build a Vault and Consul Google Image. See the [vault-consul-image example](https://github.com/hashicorp/terraform-google-vault/tree/master/examples/vault-consul-image) documentation
-   for instructions. Make sure to note down the ID of the Google Image.
+1. Build a Vault and Consul Google Image. See the [vault-consul-image example][image_example]
+  documentation for instructions. Make sure to note down the ID of the Google Image.
 1. Install [Terraform](https://www.terraform.io/).
 1. Make sure you local environment is authenticated to Google Cloud.
-1. Open `variables.tf` and fill in any variables that don't have a default, including putting your Google Image ID into
-   the `vault_source_image` and `consul_server_source_image` variables.
+1. Open `variables.tf` and fill in any variables that don't have a default, including
+  putting your Google Image ID into the `vault_source_image` and `consul_server_source_image`
+  variables.
 1. Run `terraform init`.
 1. Run `terraform plan`.
 1. If the plan looks good, run `terraform apply`.
+1. Run `curl <web_client_public_ip>:8080` to check if the web server in the client
```

Comment:
Neat!

Might want to specify that you'll need to wait a little while for the nodes to boot, unseal, bootstrap, etc.